doc/source/internals/ondisk/datetime.rst

```diff
@@ -0,0 +1,37 @@
+.. _datetime-internals:
+
+Datetime and Date
+=================
+
+Both date and datetime are stored as a unix timestamp. Datetimes are stored with microsecond precision,
+and dates merely with second precision.
+
+.. code-block:: C
+
+    struct datetime {
+        int64_t ts;
+    };
+
+    typedef datetime date;
+
+For datetimes, the timestamp is computed according to the formula::
+
+    ts = (int(timestamp) << 20) + frac(timestamp) * 1000000
+
+Examples
+--------
+
+.. code-block:: pycon
+
+    >>> buf = bytearray(1 << 20)
+    >>> end = mapped_datetime.pack_into(datetime.datetime(2019, 03, 05, 11, 15, 24, 388001), buf, 0)
```

Comment:
Check the `03` and `05`. They are octal values. Is that required?

Reply:
It's just an exaple. It's probably not necessary, but it was pasted from a console, so I'm sure it works.